bin/reth/src/commands/debug_cmd/merkle.rs

```diff
@@ -162,7 +162,7 @@ impl<C: ChainSpecParser<ChainSpec = ChainSpec>> Command<C> {
 
             td += sealed_block.difficulty();
             let mut executor = executor_provider.batch_executor(StateProviderDatabase::new(
-                LatestStateProviderRef::new(&provider_rw),
+                LatestStateProviderRef::new(provider_rw.clone()),
```

Comment:
hmm, why does this `Ref` type now requre an owned instance?

Reply:
good point, `LatestStateProviderRef` and `LatestStateProvider` can be merged with the current approach, will address this depending on how we decide to go forward